optuna/storages/_rdb/storage.py

```diff
@@ -57,11 +58,52 @@
 
     models = _LazyImport("optuna.storages._rdb.models")
 
+
+_logger = optuna.logging.get_logger(__name__)
+
+
 _RDB_MAX_FLOAT = np.finfo(np.float32).max
 _RDB_MIN_FLOAT = np.finfo(np.float32).min
 
 
-_logger = optuna.logging.get_logger(__name__)
+def _float_without_nan_to_pair_repr(value: float) -> Tuple[float, models.FloatTypeEnum]:
+    if np.isposinf(value):
+        return (0.0, models.FloatTypeEnum.INF_POS)
+    elif np.isneginf(value):
+        return (0.0, models.FloatTypeEnum.INF_NEG)
+    else:
+        return (value, models.FloatTypeEnum.FINITE_OR_NAN)
+
+
+def _float_with_nan_to_pair_repr(value: float) -> Tuple[Optional[float], models.FloatTypeEnum]:
+    if np.isnan(value):
+        return (None, models.FloatTypeEnum.FINITE_OR_NAN)
+    else:
+        return _float_without_nan_to_pair_repr(value)
+
+
+def _pair_repr_to_float_without_nan(value: float, float_type: models.FloatTypeEnum) -> float:
+
+    if float_type == models.FloatTypeEnum.INF_POS:
+        assert value == 0.0
+        return np.Infinity
```

Comment:
Could you tell me why we user `np.Infinity` here instead of built-in `inf`?
```suggestion
        return float("inf")
```

Reply:
For no particular reason. If `float("inf")` is preferrable, I'll fix it.